ts/nni_manager/training_service/local/localTrainingService.ts

```diff
@@ -236,20 +236,11 @@ class LocalTrainingService implements TrainingService {
 
             return Promise.resolve();
         }
-        tkill(trialJob.pid, 'SIGTERM');
-        const startTime = Date.now();
-        while(await isAlive(trialJob.pid)) {    
-            if (Date.now() - startTime > 4999) {
-                tkill(trialJob.pid, 'SIGKILL', (err) => {
-                    if (err) {
-                        this.log.error(`kill trial job error: ${err}`);
-                    }
-                });
-                break;
+        tkill(trialJob.pid, 'SIGKILL', (err) => {
+            if (err) {
+                this.log.error(`kill trial job error: ${err}`);
             }
-            await delay(500);
-        }
-
+        });
```

Comment:
`SIGTERM` and wait for a few time because maybe the trial has some clean logic, why this will cause the wrong job status?

Reply:
If user canceled(early stop) a trial, the status is not changed immediately, which will keep "RUNNING" for a while, From "nnimanager" main loop, there is a logic that set "RUNNING" to "FAILED" when status is "RUNNING" but the PID is not alive.

That will cause the status is "FAILED" which should be "EARLY_STOP" sometimes.

Reply:
got it, but change `SIGTERM` to `SIGKILL` may cause the early stop trial to skip its clean-up, is this a serious change? @QuanluZhang

Reply:
let's discuss it tomorrow

Reply:
updates: set early_stop before isalive check.